server/lib/pbench/process_tb.py

```diff
@@ -59,18 +58,24 @@ def _results_push(controller: str, tb: Path, token: str):
                           tar ball
             tb -- path of the tar ball
             token -- generated authorised token for Pbench user
+            agent_dir -- agent installation directory
 
         Importance of this function is while running tests we get the
-        ability to mock this function and test it easily
+        ability to mock this function and test it easily.
         """
+        pythonpath = f"{agent_dir}/lib"
+        for path in sorted(agent_dir.glob("lib*/python3.*/*")):
+            pythonpath += f":{path}"
```

Comment:
I really think (assuming that we stick with this code...) that we need to guard against adding directories from _different_ Python versions to the same `PYTHONPATH`.  Something like this might do it:
```
        if any(len(d.glob("python3.*")) > 1 for d in agent_dir.glob("lib*")):
            raise RuntimeError("Multiple Agent installations")
```
That is, for each subdirectory of `agent_dir` starting with "lib", if there is more than one subdirectory in it starting with "python3.", then raise an error.